File: backend/app/schemas/resolver.py
from typing import List, Optional

from pydantic import BaseModel, Field


class SmartResolvedEntity(BaseModel):
    entity_id: str
    entity_type: str
    name: str
    confidence: int
    reason: str
    source: str
    file_path: Optional[str] = None
    http_method: Optional[str] = None
    route_path: Optional[str] = None
    workflow_name: Optional[str] = None
    graph_connections: List[str] = Field(default_factory=list)


class ResolveRequest(BaseModel):
    query: str
    limit: int = 10


class ResolveResponse(BaseModel):
    query: str
    resolved_entities: List[SmartResolvedEntity]
    primary_entity: Optional[SmartResolvedEntity] = None
    resolution_ms: int = 0


class AutocompleteSuggestion(BaseModel):
    label: str
    entity_type: str
    entity_id: Optional[str] = None
    file_path: Optional[str] = None
    source: str
    subtitle: Optional[str] = None


class AutocompleteResponse(BaseModel):
    suggestions: List[AutocompleteSuggestion]

File: frontend/src/types/resolver.ts
export interface SmartResolvedEntity {
  entity_id: string
  entity_type: string
  name: string
  confidence: number
  reason: string
  source: string
  file_path?: string | null
  http_method?: string | null
  route_path?: string | null
  workflow_name?: string | null
  graph_connections: string[]
}

export interface ResolveResponse {
  query: string
  resolved_entities: SmartResolvedEntity[]
  primary_entity: SmartResolvedEntity | null
  resolution_ms: number
}

export interface AutocompleteSuggestion {
  label: string
  entity_type: string
  entity_id?: string | null
  file_path?: string | null
  source: string
  subtitle?: string | null
}

export interface AutocompleteResponse {
  suggestions: AutocompleteSuggestion[]
}
